### DISCUSSION TITLE
Solving moderation for cross-app comments

### DISCUSSION THREAD

**Discussion by theDanielJLewis** [Podcastindex-org/podcast-namespace#610] [Ideas]
(This is one of several posts that I'm starting to focus on solving specific issues we may have preventing full support for cross-app comments.)

Regardless of whether you agree with it, podcasters need the ability to moderate comments on their podcast and episodes.

Common needs I see:

- Removing profanity from an otherwise good comment because the podcaster insists on a family-friendly atmosphere
- Correcting legitimate but broken links—for example, someone might suggest going to `podcasting2.org` but may have accidentally typed `oodcasting2.org`
- Deleting comments for abuse, irrelevance, privacy, and such—these are cases that might not warrant completely blocking the commenter, only removing that one comment
- Deleting spam—this probably _would_ require blocking the commenter
- Sharing any of these moderation capabilities with other "team members"

(Lest you think these are all made up, I've personally performed or have seen others who needed to perform all of these moderation needs.)

All of this is possible if the comment stream is centrally hosted and editable by the podcaster.

But if the comment stream is decentralized (like ActivityPub), most of these moderation capabilities would seem impossible.

Nonetheless, I think two possible solutions for a decentralized approach would be:

1. Whatever account posted the root post is considered the moderator, or
2. Moderator-capable accounts are included in the RSS feed as the source of truth.

Then, if one of these "authorized" accounts "replies" to the problematic comment, the format of that reply can affect how that problematic comment is handled by the apps. This is similar to some email-based comment moderation tools you might have seen for blogs.

To illustrate it simply:

1. Listener1 comments: "I've loved all your other stuff, but this episode is [obscenity]!"
2. Podcaster responds: `{"type": "moderation", "action": "delete"}`
3. Comment is considered "deleted" by all apps and never shown.

Or here's an example of editing a comment:

1. Listener2 comments: "Thank you so much for this list! One more I suggest adding is Ark browser at ark.net!"
2. Podcaster response: `{"type": "moderation", "action": "edit", "content": "Thank you so much for this list! One more I suggest adding is Arc browser at https://arc.net!"}`

A problem I see with this is that while it seems like ActivityPub already has performance issues, this reply-based approach might make it even worse because it means the apps have to read the replies to comments before they can display the comments.

What other moderation needs do you foresee? And, more importantly, how can we solve these problems so that the podcaster maintains control over their property?

**Comment by Dwev** [comment]
I'm not sure commentary on a podcast is the podcaster's property, just as a review of Mission Impossible doesn't belong to Paramount. There will always be things that will and should remain out of the control of the podcaster.

All that said, I really like the idea of moderation at the host source. Although, if you reply to a comment with a moderation command using AP, won't that reply appear in the normal activity stream like Mastodon or Peertube?

You could use AP as a mechanism of _gathering_ comments from all the platforms and aggregating them at the host, and that would give you full moderation ability, as then the publishing of those comments on the feed, could be another mechanism (e.g. a simple JSON file that the apps load to display comments.). Then you could just remove a comment from the comment file, and it won't go out to the apps.  It would still be on the Activity stream though.

**Reply by theDanielJLewis** [reply]
I think comments _about_ the podcast are not the podcaster's property, but comments _on_ their podcast should be their property and under their control.

> All that said, I really like the idea of moderation at the host source. Although, if you reply to a comment with a moderation command using AP, won't that reply appear in the normal activity stream like Mastodon or Peertube?

Yes, but I believe we're looking more at the technology of ActivityPub and not necessarily Mastodon instances.

> … could be another mechanism (e.g. a simple JSON file that the apps load to display comments.)

I tried to fight that battle. No one seemed interested in a simple solution like that. 🙁

**Comment by Dwev** [comment]
A good proposal for CAC is here:

https://github.com/Podcastindex-org/podcast-namespace/discussions/623

I suggest we close this thread now and continue there, as it has developed a lot, and I think you are on board with it as well, right @theDanielJLewis?
